crates/kitsune_p2p/kitsune_p2p/src/gossip/sharded_gossip/bandwidth.rs

```diff
@@ -140,10 +139,23 @@ where
                             }
                             tokio::time::sleep(dur).await;
                         }
-                        governor::NegativeMultiDecision::InsufficientCapacity(_) => {
+                        governor::NegativeMultiDecision::InsufficientCapacity(mut cap) => {
                             tracing::error!(
-                                "Tried to send a message larger than the max message size"
+                                "Tried to send {} bits, which is larger than the maximum possible of {} bits. Allowing this large message through anyway!", bits, cap
                             );
+                            // TODO: rather than allowing this message through, we should bubble this error up so that the sender can split
+                            // the message into smaller chunks. We don't easily have that capacity right now, so, better to violate rate
+                            // limiting than to go into an infinite loop...
+
+                            // Drain the rate limiter's capacity completely, to be as accurate as possible.
+                            // (ideally we would just drain the capacity completely in one fell swoop, but `governor`'s API does not allow this.)
```

Comment:
I'm not sure I understand why we need to "drain the capacity" when we have an oversized message.

Reply:
It's because I'm allowing the message to go through even though it exceeds even the maximum possible capacity. These error cases don't actually cause any change to the state of the throttle. If there were excess capacity, it would "drain" whatever capacity it represents. So since I'm bypassing the throttle here, I'm draining any remaining capacity so that future messages still need to wait before they can pass through the now-depleted throttle. Except for another message which exceeds maximum possible capacity. But if that happens we should heed the error and make sure we are able to respond to this case and break things up so that they can pass through the throttle.

Reply:
ahhh... I was confused... I thought you were doing the opposite, trying to make it so the next message went through right away for some reason.

Ok, cool : )